src/v2/view-builder/views/shared/BaseResendView.js

```diff
@@ -32,11 +32,14 @@ export default View.extend({
     View.prototype.remove.apply(this, arguments);
     clearInterval(this.showMeInterval);
 
-    // Clear resend timeStamp whenever we change views (this means we're navigating away from the resend view)
     const formName = this.options.appState.get('currentFormName');
-    if (this.options.model 
-      && this.options.model.get('formName') !== formName
-      && sessionStorageHelper.getResendTimestamp()) {
+    const resendContext = this.options.appState.get('currentAuthenticator')?.resend 
+      || this.options.appState.get('currentAuthenticatorEnrollment')?.resend;
+    const didFormNameChange = this.options.model.get('formName') !== formName;
+
+    // Clear resend timeStamp whenever we change views (this means we're navigating away from the resend view)
+    if (sessionStorageHelper.getResendTimestamp()
+      && (!resendContext || didFormNameChange)) {
```

Comment:
would it be better to have `&&` instead of `OR`.. remove timestamp if there is no `resend` in context and form name changed.?

!resendContext && didFormNameChange

Reply:
Spoke to Pradeep about this offline, but OR is the intended operator here since we're covering the case where form name does NOT change but the new remediation does not have the "resend" context anymore